lib/producer/events_producer/exrabbitpool_connection.ex

```diff
@@ -0,0 +1,45 @@
+defmodule Mississippi.Producer.EventsProducer.ExRabbitPoolConnection do
+  @behaviour Mississippi.AMQPConnection
+
+  require Logger
+
+  def adapter(), do: ExRabbitPool.RabbitMQ
+
+  def init(state) do
+    conn = ExRabbitPool.get_connection_worker(:events_producer_pool)
+
+    with {:ok, channel} <- checkout_channel(conn),
+         :ok <- declare_events_exchange(conn, channel, state.events_exchange_name) do
+      {:ok, channel}
+    end
+  end
+
+  defp checkout_channel(conn) do
+    with {:error, reason} <- ExRabbitPool.checkout_channel(conn) do
+      _ =
+        Logger.warning(
+          "Failed to check out channel for producer: #{inspect(reason)}",
+          tag: "event_producer_channel_checkout_fail"
```

Comment:
Styler will complain of a "Missing `tag` metadata" for Logger. We can either remove the `tag` (which I did in #9 ehehe) or add extra Logger configuration (see e.g. [pretty_log](https://github.com/secomind/pretty_log))

Reply:
the tag was removed